orchestration/dagster_orchestration/hca_orchestration/tests/test_sensors.py

```diff
@@ -0,0 +1,139 @@
+import unittest
+from unittest.mock import patch
+
+from datetime import datetime
+from collections.abc import Iterable
+from typing import Generator, TypeVar
+
+from argo.workflows.client import ArchivedWorkflowServiceApi
+from argo.workflows.client.models import V1alpha1Arguments, V1alpha1Parameter, V1alpha1Workflow, V1alpha1WorkflowSpec,\
+    V1alpha1WorkflowStatus, V1ObjectMeta
+
+from hca_orchestration.contrib.argo_workflows import ExtendedArgoWorkflow, generate_argo_archived_workflows_client
+from hca_orchestration.sensors import ArgoHcaImportCompletionSensor
+
+
+T = TypeVar('T')
+
+
+# helper to turn a list into a generator to help mock generator functions
+def generator(iterable: Iterable[T]) -> Generator[T, None, None]:
+    for obj in iterable:
```

Comment:
An interesting pattern, I haven't seen this before. 👍

Reply:
it's not necessary, strictly speaking, but it's to make sure we correctly provide a generator when mocking functions that return one (e.g. the 'pull paginated results' helper)